plugins/modules/nd_federation_member.py

```diff
@@ -0,0 +1,151 @@
+    type: str
+    required: true
+    aliases: [ cluster_ip, hostname, ip_address, federation_member ]
+  cluster_username:
+    description:
+      - The username for the cluster.
+    type: str
+    required: true
+  cluster_password:
+    description:
+      - The password for the cluster.
+    type: str
+    required: true
+    no_log: true
+  cluster_login_domain:
+    description:
+      - The login domain ame to use for the cluster.
+      - Default value is set to DefaultAuth.
+    type: str
+    default: "DefaultAuth"
+  state:
+    description:
+      - The state of the cluster configuration.
+    type: str
+    default: present
+    choices: [ absent, present, query ]
+extends_documentation_fragment: cisco.nd.modules
+notes:
+- The C(federation) must exist before using this module in your playbook.
+  The M(cisco.aci.nd_federation) module can be used for this.
+"""
+
+EXAMPLES = r"""
+- name: Setup multi-cluster configuration
+  cisco.nd.nd_federation_member:
+    cluster: "172.37.20.15"
+    username: "admin"
+    password: "password"
+    login_domain: "default"
+    state: present
+"""
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        cluster=dict(type="str", aliases=["cluster_ip", "hostname", "ip_address", "federation_member"]),
+        cluster_username=dict(type="str"),
+        cluster_password=dict(type="str", no_log=True),
+        cluster_login_domain=dict(type="str", default="DefaultAuth"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["cluster"]],
+            ["state", "present", ["cluster", "cluster_username", "cluster_password"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    cluster = nd.params.get("cluster")
+    cluster_username = nd.params.get("cluster_username")
+    cluster_password = nd.params.get("cluster_password")
+    if cluster_password is not None:
+        # Make password base64 encoded
+        cluster_password = (base64.b64encode(str.encode(cluster_password))).decode("utf-8")
+    cluster_login_domain = nd.params.get("cluster_login_domain")
+    state = nd.params.get("state")
+
+    path = "/nexus/api/federation/v4/members"
+    cluster_path = path
+
+    cluster_obj = nd.query_obj(path, ignore_not_found_error=True).get("items")
+
+    if cluster_obj:
+        cluster_info = next((cluster_dict for cluster_dict in cluster_obj if cluster_dict.get("spec").get("host") == cluster), None)
+        if cluster_info:
+            cluster_path = "{0}/{1}".format(path, cluster_info.get("status").get("memberID"))
+            nd.existing = cluster_info
```

Comment:
same filter question as in federation module

Reply:
answered above